Dynatrace/dynatrace-operator
# Add hostgroup parameter to process module config request URL

## Description

Use `hostgroup` parameter for process module config request.

## How can this be tested?

1) "Create" a hostgroup  
Deploy dynakube:
```
spec:
  activeGate:
    capabilities:
      - kubernetes-monitoring
  oneAgent:
    cloudNativeFullStack:
      args:
      - "--set-host-group=host-group-test"
```
UI: 
- go to `Deployment status`/`OneAgents` page
- click `Host group: host-group-test` link below host name
- from the menu on the left choose `Container technologies`
- enable `Docker for Windows Server containers` and save changes.

2) Check value of `dockerWindowsInjection` property. It's set to `on` (after few minutes).
```
watch -n 10 /bin/bash -c "date ; kubectl -n dynatrace exec -t pod/dynatrace-oneagent-csi-driver-<HASH> -- /bin/bash -c \"cat /data/<TENANT>>/config/agent/conf/ruxitagentproc.conf\" | grep \"^.*Injection o\" | sort | grep dockerWindowsInjection"
```

3) Update dynakube:
```
spec:
  activeGate:
    capabilities:
      - kubernetes-monitoring
  oneAgent:
    cloudNativeFullStack: {}
```

4) Check value of `dockerWindowsInjection` property. It's set to `off`.

5) Switch to the `main` branch. Deploy operator from scratch. 

6) Repeat steps 1-4 (without UI part). Value of `dockerWindowsInjection` property is always `off`.

## Checklist
